src/dialogs/unblockDialogs/unblockDirectDeposit.ts

```diff
@@ -0,0 +1,349 @@
+import {
+  TextPrompt,
+  ChoicePrompt,
+  ComponentDialog,
+  WaterfallDialog
+} from 'botbuilder-dialogs';
+
+import { MessageFactory, CardFactory } from 'botbuilder';
+import i18n from '../locales/i18nConfig';
+
+import { CallbackBotDialog, CALLBACK_BOT_DIALOG } from '../callbackBotDialog';
+import { CallbackBotDetails } from '../callbackBotDetails';
+const TEXT_PROMPT = 'TEXT_PROMPT';
+const CHOICE_PROMPT = 'CHOICE_PROMPT';
+export const CONFIRM_DIRECT_DEPOSIT_STEP = 'CONFIRM_DIRECT_DEPOSIT_STEP';
+const CONFIRM_DIRECT_DEPOSIT_WATERFALL_STEP = 'CONFIRM_DIRECT_DEPOSIT_STEP';
+
+// Error handling
+const MAX_ERROR_COUNT = 3;
+let INSTITUTE = false
+let TRANSIT = false;
+const ACCOUNT = false;
+
+// WIP: EXPERIMENTING WITH ADAPTIVE CARDS IN START STEP, REMOVE LATER
+const CARD_DEBUG = null;
+
+export class UnblockDirectDepositStep extends ComponentDialog {
+  constructor() {
+    super(CONFIRM_DIRECT_DEPOSIT_STEP);
+
+    // Add a text prompt to the dialog stack
+    this.addDialog(new TextPrompt(TEXT_PROMPT));
+    this.addDialog(new ChoicePrompt(CHOICE_PROMPT));
+
+    this.addDialog(
+      new WaterfallDialog(CONFIRM_DIRECT_DEPOSIT_WATERFALL_STEP, [
+        this.unblockDirectDepositStart.bind(this),
+        this.unblockBankInstitute.bind(this),
+        this.unblockDirectDepositEnd.bind(this)
+      ])
+    );
+
+    this.initialDialogId = CONFIRM_DIRECT_DEPOSIT_WATERFALL_STEP;
+  }
+
+  /**
+   * Initial step in the waterfall. This will kick of the UnblockDirectDepositStep step
+   */
+  async unblockDirectDepositStart(stepContext) {
+
+    // Get the user details / state machine
+    const unblockBotDetails = stepContext.options;
+
+    // Check if the error count is greater than the max threshold
+    if (unblockBotDetails.errorCount.unblockDirectDeposit >= MAX_ERROR_COUNT) {
+
+      unblockBotDetails.masterError = true;
+      unblockBotDetails.unblockDirectDeposit = null;
```

Comment:
if this unblockDirectDeposit reach max error count, unblockBotDetails.unblockDirectDeposit set to -1 is better i think